### PR TITLE
Detect flow nostream/v2

### PR DESCRIPTION
**Pull request by victorjulien**
Replaces #9725
- rebase
- address type comments

### CODE REVIEW

**Review by jufajardini — COMMENTED** [review]
Whole thing mostly seems to make sense, to me. 
Nice usage of returns.
Have only two inline questions, but more out of curiosity.

File: src/detect-engine.c [hunk 1]
```diff
@@ -1990,8 +1994,8 @@ bool DetectEnginePktInspectionRun(ThreadVars *tv,
     SCEnter();
 
     for (DetectEnginePktInspectionEngine *e = s->pkt_inspect; e != NULL; e = e->next) {
-        if (e->v1.Callback(det_ctx, e, s, p, alert_flags) == false) {
-            SCLogDebug("sid %u: e %p Callback returned false", s->id, e);
+        if (e->v1.Callback(det_ctx, e, s, p, alert_flags) != DETECT_ENGINE_INSPECT_SIG_MATCH) {
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-engine.c:1997]: No, there are multiple return codes that mean "no match" for different scenarios

File: src/detect-flow.c [hunk 2]
```diff
@@ -418,6 +412,13 @@ int DetectFlowSetup (DetectEngineCtx *de_ctx, Signature *s, const char *flowstr)
     if (sm != NULL) {
         SigMatchAppendSMToList(s, sm, DETECT_SM_LIST_MATCH);
     }
+
+    if (parse_flags & DETECT_FLOW_FLAG_ONLYSTREAM) {
+        s->flags |= (SIG_FLAG_REQUIRE_STREAM | SIG_FLAG_REQUIRE_STREAM_ONLY);
+    }
+    if (parse_flags & DETECT_FLOW_FLAG_NOSTREAM) {
+        s->flags |= SIG_FLAG_REQUIRE_PACKET;
+    }
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-flow.c:421]: not specifically

**Review by jufajardini — APPROVED** [review]
Looks good to me, thanks for the answers.

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16474

**Review by inashivb — COMMENTED** [review]
One nit inline. LGTM. 🚀

**Comment by victorjulien** [comment]
Needs rebase and minor fixup.

File: src/detect-flow.c [hunk 3]
```diff
@@ -169,7 +160,8 @@ int DetectFlowMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
  * \retval fd pointer to DetectFlowData on success
  * \retval NULL on failure
  */
-static DetectFlowData *DetectFlowParse (DetectEngineCtx *de_ctx, const char *flowstr)
+static DetectFlowData *DetectFlowParse(
+        DetectEngineCtx *de_ctx, const char *flowstr, uint16_t *parse_flags)
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/detect-flow.c:164]: added

**Comment by victorjulien** [comment]
replaced by #9798

**Pull request closed without merging**